### PR TITLE
AP-2883 employment integration tests

### PR DESCRIPTION
**Pull request by stephenrichards**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2883)

Add method to create remarks for multiple employments and tax and ni refunds
revise calculation of disposable income to not double count gross_employment_income
Add method to calculate blunt average when we cannot determine the payment period
Update period analyzer to include 5 different payments for 28day pay peeriods to reflect data from HMRC
And tests for the new methods 
Update google sheet to show expected remarks on tests 1,3 and 5
https://docs.google.com/spreadsheets/d/1fv0oNuQBIB_r2krsRCAM8lOVKdkaf8olbr6Wa-cVEE0/edit#gid=1793424564

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/services/collators/gross_income_collator.rb [hunk 1]
```diff
@@ -33,6 +37,7 @@ def populate_income_attrs(attrs, category)
 
     def default_attrs
       {
+        # this looks wrong, monthly_student_loan is used for total gross income
```
Comment (chris-groves) [inline app/services/collators/gross_income_collator.rb:40]: is there a separate ticket to look at this?

File: lib/integration_helpers/test_case/v4/remarks_comparer.rb [hunk 2]
```diff
@@ -33,21 +33,34 @@ def compare_remarks
...
       def compare_remark_type(type, hash)
         hash.each do |issue, _ids|
+          puts "before check"
+          print_remark_line(type, issue)
           next if @actual&.dig(type)&.dig(issue) == @expected[type][issue]
 
           @result = false
-          print_remark_line(type, issue) unless silent?
+          print_remark_line(type, issue)
+          # print_remark_line(type, issue) unless silent?
```
Comment (chris-groves) [inline lib/integration_helpers/test_case/v4/remarks_comparer.rb:42]: can this be removed?

File: lib/integration_helpers/test_case/v4/remarks_comparer.rb [hunk 3]
```diff
@@ -33,21 +33,34 @@ def compare_remarks
...
 
       def print_remark_line(type, issue)
-        color = :red
-        puts "#{type}/#{issue}".__send__(color)
-        puts "  expected:"
-        @expected[type][issue].each { |id| puts "    #{id}" }
-        puts "  actual:"
-        @actual&.dig(type)&.dig(issue)&.each { |id| puts "    #{id}" }
-        puts "  "
+        # # OLD STYLE
```
Comment (chris-groves) [inline lib/integration_helpers/test_case/v4/remarks_comparer.rb:47]: can this comment be removed?

File: lib/integration_helpers/test_case/v4/remarks_comparer.rb [hunk 4]
```diff
@@ -33,21 +33,34 @@ def compare_remarks
...
+
+        # # NEW STYLE
+        # color = :red
+        # puts "#{type}/#{issue}".__send__(color)
+        # puts "  expected:"
+        # @expected[type][issue].each { |id| puts "    #{id}" }
+        # puts "  actual:"
+        # @actual&.dig(type)&.dig(issue)&.each { |id| puts "    #{id}" }
+        # puts "  "
       end
```
Comment (chris-groves) [inline lib/integration_helpers/test_case/v4/remarks_comparer.rb:64]: can this commented out code be removed?

File: lib/integration_helpers/test_case/v4/result_comparer.rb [hunk 5]
```diff
@@ -30,7 +30,8 @@ def call
     private
 
       def silent?
-        @verbosity.zero?
+        # @verbosity.zero?
```
Comment (chris-groves) [inline lib/integration_helpers/test_case/v4/result_comparer.rb:33]: can this comment be removed?

File: app/services/collators/gross_income_collator.rb (hunk 1 above)
Comment (stephenrichards) [reply app/services/collators/gross_income_collator.rb:40]: It does look a bit odd, but in fact is correct.  It's setting up the initial values of `attrs` here, and then adds in for each category on line 25.  I've changed Will's comment to read `# setup initial values here`

**Review by mpw5 — APPROVED** [review]
LGTM

**Pull request merged**